ui/appui-react/src/appui-react/frontstage/FrontstageDef.tsx

```diff
@@ -674,6 +697,39 @@ export class FrontstageDef {
     this._bottomMostPanel = Frontstage.createStagePanelDef(StagePanelLocation.BottomMost, props);
   }
 
+  /** Initializes a FrontstageDef from FrontstageConfig.
+   * @internal
+   */
+  public async initializeFromConfig(config: FrontstageConfig): Promise<void> {
+    this._id = config.id;
+    this._initialConfig = config;
+    this._defaultTool = config.defaultTool;
+
+    if (config.defaultContentId !== undefined)
+      this._defaultContentId = config.defaultContentId;
+
+    if (config.contentGroup instanceof ContentGroupProvider) {
+      this._contentGroupProvider = config.contentGroup;
+    } else {
+      this._contentGroup = config.contentGroup;
+    }
+
+    if (config.applicationData !== undefined)
+      this._applicationData = config.applicationData;
+
+    this._usage = config.usage;
+    this._version = config.version || 0;
+
+    this._toolSettings = createWidgetDef(config.toolSettings, `uifw-toolSettings-widget`, WidgetType.ToolSettings);
+    this._statusBar = createWidgetDef(config.statusBar, `uifw-statusBar-widget`, WidgetType.StatusBar);
+    this._contentManipulation = createWidgetDef(config.contentManipulation, `uifw-contentManipulation-widget`, WidgetType.Tool);
+    this._viewNavigation = createWidgetDef(config.viewNavigation, `uifw-viewNavigation-widget`, WidgetType.Navigation);
```

Comment:
I think these should still populate the "zone" version of this, otherwise the fronstage is branching and that might break code that still use the previous stuff, I dont know to which extent it would cause a problem. (Could a package provided frontstage be interracting with an application that would not be updated and still look at "topLeft" instead of "contentManipulation" ?

Reply:
You are right, that conversion step makes sense, especially when gradually reacting to the changes in 3.x. Will work on that.

Reply:
Refactored as suggested.